File: typescript/src/restClient.ts
import { performance } from "perf_hooks";
import axios from "axios";

const baseURL = "http://localhost:5000";

const testCounts = [100, 200, 300];

const endpoints = {
  usuarios: "/usuarios",
  musicas: "/musicas",
  playlists: "/playlists",
};

async function testLoad(endpoint: string, name: string, numTest: number) {
  const url = `${baseURL}${endpoint}`;
  const start = performance.now();

  for (let i = 0; i < numTest; i++) {
    await axios
      .get(url)
      .catch((error) => console.error(`Error: ${error.message}`));
  }

  const end = performance.now();
  console.log(
    `${name} load test with ${numTest} requests completed in ${
      (end - start) / 1000
    } seconds`
  );
}

export async function runTests() {
  for (let numTest of testCounts) {
    console.log(`Running tests with ${numTest} requests:`);
    await testLoad(endpoints.usuarios, "Usuarios", numTest);
    await testLoad(endpoints.musicas, "Musicas", numTest);
    await testLoad(endpoints.playlists, "Playlists", numTest);
  }
}

File: rust/src/rest_client.rs
use reqwest::Client;
use std::time::Instant;

const BASE_URL: &str = "http://localhost:5000";

pub async fn run_tests() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new();
    let test_counts = vec![100, 200, 300];
    let endpoints = vec!["/usuarios", "/musicas", "/playlists"];

    for &count in &test_counts {
        println!("Running tests with {} requests:", count);
        for &endpoint in &endpoints {
            let start = Instant::now();
            for _ in 0..count {
                let url = format!("{}{}", BASE_URL, endpoint);
                let res = client.get(&url).send().await;
                if let Err(err) = res {
                    eprintln!("Request failed: {:?}", err);
                }
            }
            let duration = start.elapsed();
            println!(
                "{} load test with {} requests completed in {:.2?} seconds",
                endpoint, count, duration
            );
        }
    }

    Ok(())
}
